packages/material-ui/src/styles/index.d.ts

```diff
@@ -24,16 +24,18 @@ export {
   StyledComponentProps,
 } from './withStyles';
 export { default as withTheme, WithTheme } from './withTheme';
-export { default as muiStyled } from './muiStyled';
+export { default as experimentalStyled, CreateMUIStyled } from './experimentalStyled';
 export { default as styled, ComponentCreator, StyledProps } from './styled';
+export {
+  default as MuiThemeProvider,
```

Comment:
We have stopped documenting `MuiThemeProvider` in v4. I would propose we keep pushing in the same direction: we drop `MuiThemeProvider` in v5. The motivation for using `ThemeProvider ` vs. `MuiThemeProvider ` is to signal that you only need one theme provider in the page, not one from Material-UI and one from styled-components. I have created https://trello.com/c/q6cdyyaP/2744-remove-muithemeprovider to keep track of this proposal.